airflow/serialization/serialized_objects.py

```diff
@@ -409,30 +406,59 @@ def _value_is_hardcoded_default(cls, attrname: str, value: Any, instance: Any) -
             return True
         return False
 
+    @classmethod
+    def _serialize_param(cls, param: Param):
+        return dict(
+            __class=f"{param.__module__}.{param.__class__.__name__}",
+            default=cls._serialize(param.value),
+            description=cls._serialize(param.description),
```

Comment:
Do we really need to serialize/deserialize a parma's description as well? It ought to be a simple string only.

Reply:
So I think it ends up being the same because string is primitive type

Reply:
yeah just double checked and when it is primitive type, `_serialize` doesn't do anything

though, maybe we should not serialize it so as not to accidentally support something users should not be doing anyway.  so on second thought, probably should change this to put the raw value  (in which case, json-serialization will fail)

Reply:
ok just sat down to not `_serialize` the `description` attr.  and on second thought, i think it's better to run it through `_serialize` after all.  like, that's precisely what `_serialize` is there for.  there's a reason that `_serialize` has handling for primitive types, and so we should use it; to do otherwise is essentially to duplicate serialization logic.  not running through `_serialize` would not prevent a user from using other json-serializable types (only would prevent from using non-json-ser types like `set`) -- if we really want to clamp down on types, probably the right place do do that is in `Parame.__init__`.

so, i think i'll leave this as is, but let me know if you think otherwise.

thanks